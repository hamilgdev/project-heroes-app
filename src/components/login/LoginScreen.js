import { useNavigate } from "react-router-dom";

const LoginScreen = () => {
  const navigate = useNavigate();

  const handleLogin = () => {
    navigate("/marvel", { replace: true });
  };

  return (
    <div className="flex items-center justify-center min-h-screen p-4">
      <div className="w-full text-center">
        <div className="mb-4 | text-gray-600">
          <h2 className="font-bold uppercase">Login</h2>
          <hr className="text-gray-300" />
        </div>
        <button
          className="px-2 py-1 | bg-sky-900 text-sky-300 | rounded-md | hover:bg-sky-300 hover:text-sky-900"
          onClick={handleLogin}
        >
          Login
        </button>
      </div>
    </div>
  );
};

export default LoginScreen;
