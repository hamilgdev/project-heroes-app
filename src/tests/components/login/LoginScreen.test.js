import { mount } from "enzyme";
import { MemoryRouter, Route, Routes } from "react-router-dom";
import { AuthContext } from "../../../auth/authContext";
import LoginScreen from "../../../components/login/LoginScreen";
import { types } from "../../../types";

const mockNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockNavigate,
}));

describe("Tests in <LoginScreen />", () => {
  const contextValue = {
    dispatch: jest.fn(),
    user: { logged: false },
  };

  const wrapper = mount(
    <AuthContext.Provider value={contextValue}>
      <MemoryRouter initialEntries={["/login"]}>
        <Routes>
          <Route path="/login" element={<LoginScreen />} />
        </Routes>
      </MemoryRouter>
    </AuthContext.Provider>
  );

  test("should display correctly by default", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should to the dispatch and navigate", () => {
    wrapper.find("button").prop("onClick")();
    expect(contextValue.dispatch).toHaveBeenCalledWith({
      type: types.login,
      payload: { name: "Welcome" },
    });
    expect(mockNavigate).toHaveBeenCalledWith("/marvel", { replace: true });
  });
});
