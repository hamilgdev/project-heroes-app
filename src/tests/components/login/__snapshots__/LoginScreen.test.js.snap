// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <LoginScreen /> should display correctly by default 1`] = `
<div
  className="flex items-center justify-center min-h-screen p-4"
>
  <div
    className="w-full text-center"
  >
    <div
      className="mb-4 | text-gray-600"
    >
      <h2
        className="font-bold uppercase"
      >
        Login
      </h2>
      <hr
        className="text-gray-300"
      />
    </div>
    <button
      className="px-2 py-1 | bg-sky-900 text-sky-300 | rounded-md | hover:bg-sky-300 hover:text-sky-900"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;
